pkg/sql/sem/tree/datum.go

```diff
@@ -71,6 +71,7 @@ var (
 // Datum represents a SQL value.
 type Datum interface {
 	TypedExpr
+	json.JSONizable
```

Comment:
Consider adding a comment above this explaining that it's needed for the `to_jsonb` function, which needs to be able to convert arbitrary datums into JSON.